storage/innobase/log/log0recv.cc

```diff
@@ -1830,6 +1830,16 @@ static byte *recv_parse_or_apply_log_rec_body(
             !fsp_is_system_or_temp_tablespace(space_id) &&
             /* For cloned db header page has the encryption information. */
             !recv_sys->is_cloned_db) {
+          /* If Encryption info is changed after the last checkpoint lSN, scan
+          the tablespace to know modified pages. This scan and copy of modified
+          pages would avoid the apply the redo logs due to changed in encryption
+          info. As there is problem of apply redo on page 0 because it reset
+          in-memory encryption and page can't decrypted */
```

Comment:
```suggestion
          /* If tablespace has been in-place encrypted ie ALTER TABLESPACE
          ENCRYPTION=Y/N after the last checkpoint LSN (after full backup),
          and if the checkpoint LSN doesn't move between full and incremental
          backups, page tracking will not give the full list of modified pages. So
          we rely on the redo to roll forward the changes.
          
          With redo based apply, if page 0 redo is applied first, it will remove
          the encryption info and after this, other encrypted pages in tablespace
          cannot be decrypted.
          
          Scan and copy modified pages of such tablespaces during the incremental
          backup. This way we would have the latest state of tablespace and redo apply
          will skip all the redo generated on the tablespace. */
```